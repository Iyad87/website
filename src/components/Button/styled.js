import styled from 'styled-components';
import { highlight, textOnDark } from '../../styles/colours';

export const withButtonStyles = Component => styled(Component)`
  background: ${highlight};
  box-shadow: 0 25px 30px 5px rgba(88, 86, 214, 0.2);
  color: ${textOnDark};
  display: inline-block;
  font-size: 1.125em;
  font-weight: 400;
  text-align: center;
  text-decoration: none;
  min-width: 10em;
  padding: 20px;
`;
