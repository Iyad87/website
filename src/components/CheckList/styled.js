import styled from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import breakpoints from '../../styles/breakpoints';
import { spacing } from '../../styles/spacing';
import { textLight } from '../../styles/colours';

export const List = styled.ul`
  display: flex;
  flex-wrap: wrap;
  list-style-type: none;
  margin: ${spacing(6)} 0;
  padding: 0;
`;

export const ListItem = styled.li`
  color: ${textLight};
  display: flex;
  flex-direction: row;
  font-size: 1.25rem;
  margin: ${spacing(2)} 0;
  width: 100%;

  ${breakpoints.tablet`
    padding-right: ${spacing(2)};
    width: 50%;

    &:nth-child(2n) {
      padding-left: ${spacing(2)};
      padding-right: 0;
    }
  `};
`;

export const Icon = styled(FontAwesomeIcon)`
  color: #37beb0;
  font-size: 2rem;
  margin-right: ${spacing(3)};
`;
