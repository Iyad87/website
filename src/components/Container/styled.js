import styled from 'styled-components';

import { gridWidth, gridWidthWide, gridColWidth } from '../../styles/spacing';

const maxWidth = {
  normal: `${gridWidth + gridColWidth}px`,
  wide: `${gridWidthWide + gridColWidth}px`,
};

export const StyledContainer = styled.div`
  margin: 0 auto;
  padding: 0 ${gridColWidth / 2}px;
  max-width: ${props => (props.wide ? maxWidth.wide : maxWidth.normal)};
`;
