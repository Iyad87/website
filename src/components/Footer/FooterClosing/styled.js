import styled from 'styled-components';

import DefaultLink from 'components/Link';
import { footerText } from 'styles/colours';
import { spacing } from 'styles/spacing';
import breakpoints from 'styles/breakpoints';

export const Container = styled.section`
  color: ${footerText};
  display: flex;
  flex-direction: column-reverse;
  text-align: center;

  ${breakpoints.desktop`
    flex-direction: row;
    justify-content: space-between;
  `};
`;

export const Copyright = styled.p`
  margin-bottom: ${spacing()};
  margin-top: ${spacing()};
`;

export const Link = styled(DefaultLink)`
  color: ${footerText};
  font-weight: 400;
`;

export const SiteLinks = Copyright.withComponent('ul').extend`
  list-style-type: none;
  padding: 0;
`;

export const SiteLink = styled.li`
  display: inline-block;
  padding-left: ${spacing()};
  padding-right: ${spacing()};

  ${breakpoints.desktop`
    &:first-child {
      padding-left: 0;
    }

    &:last-child {
      padding-right: 0;
    }
  `};
`;

export const SocialLinks = styled.ul`
  list-style-type: none;
  margin: 0;
  padding: 0;
`;

export const SocialLink = SiteLink.extend`
  font-size: 2rem;
`;
