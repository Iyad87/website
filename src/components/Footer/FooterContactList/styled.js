import styled from 'styled-components';

import DefaultLink from 'components/Link';
import { footerText } from 'styles/colours';
import { spacing } from 'styles/spacing';
import breakpoints from 'styles/breakpoints';

export const Contact = styled.li`
  padding: ${spacing(2)};
  width: 50%;

  ${breakpoints.tablet`
    margin-bottom: 0;
    width: ${100 / 3}%;
  `};
`;

export const List = styled.ul`
  display: flex;
  flex-wrap: wrap;
  list-style-type: none;
  margin: -${spacing(2)};
  padding: 0;
`;

export const Line = styled.p`
  margin-bottom: ${spacing()};
  margin-top: 0;
  font-size: 1.125rem;
  font-style: normal;
`;

export const Link = styled(DefaultLink)`
  color: ${footerText};
  font-weight: 400;
`;

export const Office = Line.withComponent('h6').extend`
  font-weight: 500;
  margin-bottom: ${spacing(2)};
`;
