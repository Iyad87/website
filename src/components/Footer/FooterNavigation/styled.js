import styled from 'styled-components';

import DefaultLink from 'components/Link';
import { footerText } from 'styles/colours';
import { spacing } from 'styles/spacing';

export const Category = styled.ul`
  list-style-type: none;
  margin: 0;
  padding: 0;
  width: 25%;
`;

export const Item = styled.li`
  color: ${props => (props.title ? 'FFFFFF' : footerText)};
  margin-bottom: ${spacing()};
  margin-top: 0;
  font-size: 1.125rem;
  font-style: normal;
  font-weight: ${props => (props.title ? 500 : 400)};
`;

export const Nav = styled.div`
  display: flex;
  margin-top: ${spacing(5)};
`;

export const Link = styled(DefaultLink)`
  color: ${footerText};
  font-weight: 400;
`;
