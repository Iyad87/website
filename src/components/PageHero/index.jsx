import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { DefaultTitle, Main } from './styled';

const PageHero = ({ title, children }) => {
  const renderedTitle =
    typeof title === 'string' ? <DefaultTitle>{title}</DefaultTitle> : title;

  return (
    <Fragment>
      <header>{renderedTitle}</header>
      <Main>{children}</Main>
    </Fragment>
  );
};

PageHero.propTypes = {
  children: PropTypes.node,
  title: PropTypes.node.isRequired,
};

PageHero.defaultProps = {
  children: [],
};

export default PageHero;
