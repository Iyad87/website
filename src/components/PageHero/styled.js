import styled from 'styled-components';
import { textLight } from 'styles/colours';
import { gridWidth, spacing } from 'styles/spacing';

export const DefaultTitle = styled.h1`
  font-size: 3rem;
  font-weight: 500;
  color: ${textLight};
  line-height: 1em;
  margin-bottom: ${spacing(2)};
  margin-top: ${spacing(4)};
`;

export const Main = styled.main`
  color: ${textLight};
  font-size: 1.25rem;
  line-height: 2em;
  max-width: ${gridWidth * 2 / 3}px;
`;
