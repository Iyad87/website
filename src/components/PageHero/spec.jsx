import React from 'react';
import { shallow } from 'enzyme';
import styled from 'styled-components';
import PageHero from './index';

describe('components/PageHero', () => {
  test('basic snapshot', () => {
    const wrapper = shallow(
      <PageHero title="Hello, world!">I am the page hero body</PageHero>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('renders children', () => {
    const children = 'I am the page hero body';

    const wrapper = shallow(
      <PageHero title="Hello, world!">{children}</PageHero>,
    );

    expect(
      wrapper
        .children()
        .find('styled(main)')
        .dive()
        .text(),
    ).toEqual(children);
  });

  test('renders DefaultTitle if only a string is provided as title', () => {});

  test('renders title as a component if it is not a string', () => {});
});
