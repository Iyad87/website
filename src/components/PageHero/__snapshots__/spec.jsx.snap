// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PageHero basic snapshot 1`] = `
<React.Fragment>
  <header>
    <styled.h1>
      Hello, world!
    </styled.h1>
  </header>
  <styled.main>
    I am the page hero body
  </styled.main>
</React.Fragment>
`;

exports[`compositions/homepage/PageHero basic snapshot 1`] = `
<React.Fragment>
  <header>
    <styled.h1 />
  </header>
  <styled.main />
</React.Fragment>
`;

exports[`compositions/homepage/PageHero snapshot 1`] = `
<React.Fragment>
  <header />
  <styled.main />
</React.Fragment>
`;
